backend/app/controllers/api/Blobs.scala

```diff
@@ -88,13 +90,39 @@ class Blobs(override val controllerComponents: AuthControllerComponents, manifes
   }
 
 
-  def delete(id: String, checkChildren: Boolean): Action[AnyContent] = ApiAction.attempt { req =>
+  def delete(id: String, checkChildren: Boolean, isAdminDelete: Boolean): Action[AnyContent] = ApiAction.attempt { req =>
     import scala.language.existentials
     val deleteResource = new DeleteResource(manifest, index, previewStorage, objectStorage, postgresClient)
-    checkPermission(CanPerformAdminOperations, req) {
-      val result = if (checkChildren) deleteResource.deleteBlobCheckChildren(id)
-      else deleteResource.deleteBlob(id)
-      result.map(_ => NoContent)
+    if (isAdminDelete) {
+      checkPermission(CanPerformAdminOperations, req) {
+        val result = if (checkChildren) deleteResource.deleteBlobCheckChildren(id)
+        else deleteResource.deleteBlob(id)
+        result.map(_ => NoContent)
+      }
+    } else {
+      deleteForNoneAdmin(req.user, id).map(_ => NoContent)
     }
   }
+
+  private def deleteForNoneAdmin(user: User, blobUri: String): Attempt[Unit] = {
+    manifest.getCollectionsForBlob(blobUri).flatMap { collections =>
+      // Here we check either of 2 followings:
+      // if there's only 1 collection holding the blob and if the requesting user has view access to the collection
+      // OR if the requesting user is the only creator of the blob
```

Comment:
maybe extend this '...creator of the blob e.g. if a user has uploaded the same file to multiple workspaces'